controlplane/kubeadm/api/v1beta1/kubeadm_control_plane_webhook.go

```diff
@@ -152,6 +152,7 @@ func (in *KubeadmControlPlane) ValidateUpdate(old runtime.Object) error {
 		{spec, "machineTemplate", "infrastructureRef", "apiVersion"},
 		{spec, "machineTemplate", "infrastructureRef", "name"},
 		{spec, "machineTemplate", "nodeDrainTimeout"},
+		{spec, "machineTemplate", "nodeDeletionTimeout"},
```

Comment:
Can you please also extend the unit test?

You only have to set a value in ~l.242 and then change it in ~367.

We just had a few cases in the past where we thought something is mutable, but it wasn't.